Auth.py

```diff
@@ -299,4 +306,78 @@ def blacklist_token(self, token):
+
+        cursor.execute("DELETE FROM users WHERE (firstname = 'Test' AND lastname = 'Test' );")
+
+        conn.commit()
+        conn.close()
+            
+        error_messages = ["empty-first-name", "empty-last-name", "empty-email", 
+                              "email-already-taken", "invalid-email", 
+                              "password-too-short", "passwords-no-match"]
+        result = self.sign_up( firstname="Test", lastname="Test", email="test@gmail.com", password="longpassword", confirmpassword="longpassword")
+
+        if result in error_messages:
+            print("Something went wrong in the sign_up_test, the error was: " + result)
+        else:
+            print("Everything went fine in the sign_up_test")
+        #These functions run the individual tests
+        
+
+    def sign_in_test(self):
+        error_messages = ["user-no-exist", "incorrect-email-or-password"]
+
+        result = self.sign_in( email="test@gmail.com", password="longpassword")
+
+        if result in error_messages:
+            print("Something went wrong in the sign_in_test, the error was: " + result)
+        else: 
+            print("Everything went fine in the sign_in_test")
+        
+
+    def validate_token_test(self):
+        error_message = ["ERROR_INVALID_TOKEN"]
+
+        result = self.validate_token( token="eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJoaWdobG93IiwiaWF0IjoxNTU2MTUxMzI0LjAsInN1YiI6IjM3ZTIxZjI2LTY2ZWYtMTFlOS04YTliLTAyNDJhYzExMDAwMiIsImV4cCI6MTU3MTkxOTMyNC4wfQ.s2payesaWIfMMNwF0ofNHsxSyGACFRVhj3gh4C8AQp8")
+
+        if len( result.split(".") ) == 3:
+            if result not in error_message:
+                print("Everything went fine in the validate_token_test")  
+            else:
+                print("Something went wrong in the validate_token_test, the error was: " + result)    
+        
+
+    def send_password_reset_email_test(self):
+        result = self.send_password_reset_email( email="test@gmail.com")
+
+        if result == "success":
+            print("send_password_reset_email was a success")
+        else:
+            print("send_password_reset_email was not a success, the error is: " + result)
+        
+
+    def reset_password_test(self):
+        result = self.reset_password( token="eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJoaWdobG93IiwiaWF0IjoxNTU2MTUxMzI0LjAsInN1YiI6IjM3ZTIxZjI2LTY2ZWYtMTFlOS04YTliLTAyNDJhYzExMDAwMiIsImV4cCI6MTU3MTkxOTMyNC4wfQ.s2payesaWIfMMNwF0ofNHsxSyGACFRVhj3gh4C8AQp8",
+                                        password="longpassword", confirmpassword="longpassword")
+            
+        error_messages = ["ERROR-INVALID_TOKEN", "passwords-no-match"] 
+            
+        if result in error_messages:
+            print("Something went wrong in the reset_password_test, the error is " + result)
+        elif result == "success":
+            print("Everything went fine in the reset_password_test")
+        
+        #Remove the user that we previously added
```

Comment:
Ok, now that I get here I understand why you didn't delete it in the `sign_up` function - because then you couldn't test any of the other ones.

However, since this chunk of code really has nothing to do with resetting the password, there are two main options: (1) move it out of this function and into the `__init__`, or (2) **Just delete it. It's not really necessary since we delete it at the beginning anyway.**

Reply:
Lastly, for these functions, it's okay not to use the parameter labels. It'll still work, but since they were defined internally as positional arguments, it ignores the labels, so even if you had the wrong labels, it would treat them based on their positions and still work, which could be confusing.

Reply:
Yeah, the only reason I put those parameters in is because it was throwing an error on my computer and I forgot to remove them. Thanks!